dali/operators/image/crop/bbox_crop.cc

```diff
@@ -525,22 +525,25 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
     }
     tp.RunAll();
 
-    auto &anchor_out = ws.template OutputRef<CPUBackend>(0);
-    anchor_out.Resize(uniform_list_shape(num_samples, {ndim}));
-    auto anchor_out_view = view<float>(anchor_out);
+    DALIDataType output_dtype = has_crop_shape_ ? DALI_INT32 : DALI_FLOAT;
```

Comment:
I wonder if this is a good idea to change the output type based on presence of one of the arguments.
Maybe I always should be DALI_FLOAT? Or just DALI_INT32 and use cast operator after that operator?

Reply:
I think that if user provided crop_shape argument (which represents the exact size in pixels of desired crop region) then output anchors and shapes should also be returned as integers. Returning them as relative float values could potentially result in output shape being off by 1 pixel due to fp arithmetic errors and this behaviour could break the pipeline in some use cases.

Reply:
How about returning always FLOAT in that case (but with switch that is by default on that changes it to int - so we won't break the existing code while introducing new functionality).

Reply:
This requirement for different output type depending on whether crop_shape argument was provided or not is imposed because of the way Slice operator interprets its input arguments (int -> pixel coords, float -> normalized coords). I believe that current fix won't break any existing code - any previous usage of RandomBBoxCrop operator with crop_shape provided required additional cast to int to be declared explicitly in pipeline - current fix will provide this cast unnecessary, although the results will be the same.

Reply:
The only other solution I see is to always return int values (in pixel coordinates), even when using scaling instead of crop_shape (possibly with switch to return the output in normalized coords instead). However, this will probably require much more existing code to be rewritten.

Reply:
There's another problem. We cannot return pixel coordinates with scaling provided as we lack image shape data.

Reply:
I think that returning different types should be OK, given that `Slice` operator interprets those differently.

Reply:
We've discussed internally with the team and came to the conclusion that changing the output type of the operator silently, based on the presence of another argument, is not a good user experience, since it could break some existing code. So, for the sake of backward compatibility we should keep DALI_FLOAT by default.

I can suggest some solutions:
1. Use `normalized_anchor=False`, `normalized_shape=False` in Slice, so that you don't need to cast to int32
2. Cast to int32 (it adds extra overhead, not recommended)
3. Add a output_dtype argument to this operator, and only use int32 when it was explicitly requested.

Reply:
Personally, I've found pixel coordinates returned as a DALI_FLOAT confusing, because of the way `Slice` works by default, although I understand your point. Currently I've reverted my changes (leaving only the sigsegv fix). Maybe additional clarification should be added to `RandomBBoxCrop` docs?